src/sqs.ts

```diff
@@ -113,11 +129,49 @@ export class SQSMessageHandler<Message, Context> {
             await action({ ...context, logger: messageLogger }, parsedMessage);
           }
 
-          messageLogger.info('Successfully processed message');
+          messageLogger.info('Successfully processed SQS message');
         },
       );
 
-      context.logger.info('Succesfully processed all messages');
+      if (!processingResult.hasUnprocessedRecords) {
+        context.logger.info('Successfully processed all SQS messages');
+      }
+
+      if (!this.config.usePartialBatchResponses) {
+        processingResult.throwOnUnprocessedRecords();
+        return;
+      }
+
+      // SQS partial batching expects that you return an ordered list of
+      // failures. We map through each group and add them to the batch item
+      // failures in order for each group.
+      const batchItemFailures = Object.entries(
+        processingResult.unprocessedRecords,
+      )
+        .map(([groupId, record]) => {
+          const [failedRecord, ...subsequentUnprocessedRecords] = record.items;
+          context.logger.error(
+            {
+              groupId,
+              err: record.error,
+              failedRecord,
+              subsequentUnprocessedRecords,
```

Comment:
Will the current logs obfuscation feature also mask these logged records, or will additional logic now be needed in the obfuscation code?

Reply:
Good observation. I'm taking a closer look at this now.

Reply:
I actually noticed we were already logging the full event at the very beginning. @swain I know we use this for debugging, but it definitely seems like a big issue that we're potentially exposing PHI. I'm thinking of temporarily removing logging `event.Records`. Thoughts?

Reply:
Given we were already logging the records, I'm at odds. I'm thinking of merging this as is and coming back to add a universal obfuscation solution.

Reply:
I think I'd agree that obfuscating the logging of the SQS handler is out-of-scope, since that handler doesn't support obfuscation anyway. It definitely should, but I don't think that's "breaking" in this PR.

The Dynamo handler _does_ support obfuscation. If that behavior is breaking in any way, we should fix it here.

Reply:
Ah I forgot that only the DynamoDB handler supports obfuscation. Sounds like a great plan.